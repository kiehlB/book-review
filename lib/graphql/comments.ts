import gql from 'graphql-tag';

export const CreateComment = gql`
  mutation CreateComments($post_id: String!, $text: String!, $comment_id: String) {
    createComments(post_id: $post_id, text: $text, comment_id: $comment_id) {
      id
      level
    }
  }
`;

export const RemoveComments = gql`
  mutation RemoveComments($id: String!) {
    removeComments(id: $id)
  }
`;

export const GET_COMMENTS_COUNT = gql`
  query GetCommentsCount($id: String!) {
    post(id: $id) {
      id
      comments_count
    }
  }
`;

export const EditComments = gql`
  mutation EditComments($id: String!, $text: String!) {
    editComments(id: $id, text: $text) {
      id
      text
      level
      updated_at
      created_at
      updated_at
    }
  }
`;

export const GET_SubComment = gql`
  query GetSubComments($comment_id: String!) {
    getSubComments(comment_id: $comment_id) {
      id
      text
      likes
      has_replies
      deleted
      upvotes
      likedComments
      users {
        id
        username
      }
      level
      reply
      created_at
      updated_at
      replies_count
      replies {
        id
        text
        level
        updated_at
        created_at
        has_replies
        replies_count
        upvotes
        users {
          id
          username
          profile {
            id
            bio
            profile_name
            thumbnail
          }
        }
      }
    }
  }
`;

export const CommentUpvotes = gql`
  mutation commentUpvotes($id: String!) {
    commentUpvotes(id: $id)
  }
`;

export const DeleteCommentUpvote = gql`
  mutation DeleteCommentUpvote($id: String!) {
    deleteCommentUpvote(id: $id)
  }
`;
