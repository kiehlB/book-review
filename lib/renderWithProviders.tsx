import React from 'react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';

type Options = {
  mocks?: MockedResponse[];
  routeOptions?: {
    route: string;
  };
};

export default function renderWithProviders(
  ui: React.ReactNode,
  { mocks, routeOptions }: Options = {},
) {
  return {
    ...render(
      <HelmetProvider>
        <MockedProvider mocks={mocks} addTypename={false}>
          {ui}
        </MockedProvider>
      </HelmetProvider>,
    ),
  };
}
