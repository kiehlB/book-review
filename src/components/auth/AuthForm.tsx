import * as React from 'react';
import { motion } from 'framer-motion';
import { ApolloError } from '@apollo/client';
import LabelInput from '../common/LabelInput';
import clsx from 'clsx';
import Link from 'next/link';
import { Input, useInput, Grid } from '@nextui-org/react';
import Google from '../../svg/google';
import FaceBook from '../../svg/facebook';
import { toast } from 'react-toastify';
import styled from 'styled-components';
import useDarkMode from '../base/useDarkmode';

export interface inputProps {
  password: string | number | readonly string[];
  email: string | number | readonly string[];
}

export declare type BindingsChangeTarget =
  | React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  | string;

export interface AuthFormProps {
  email?: string;
  password?: string;
  EB: {
    value: string;
    onChange: (event: BindingsChangeTarget) => void;
  };
  PB: {
    value: string;
    onChange: (event: BindingsChangeTarget) => void;
  };
  helper: {
    color: 'error' | 'default' | 'primary' | 'secondary' | 'success' | 'warning';
    text: 'error' | 'default' | 'primary' | 'secondary' | 'success' | 'warning';
  };
  handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  handleSubmit?: (e: React.FormEvent<HTMLFormElement>) => void;
  authError?: ApolloError;
  auth?: string;
  isRegister?: string;
  linkTo?: string;
  Passwordhelper: {
    color: 'error' | 'default' | 'primary' | 'secondary' | 'success' | 'warning';
    text: 'error' | 'default' | 'primary' | 'secondary' | 'success' | 'warning';
  };
  mode: string;
  error: ApolloError;
}

const AuthForm: React.FC<AuthFormProps> = ({
  handleSubmit,
  authError,
  auth,
  EB,
  PB,
  helper,
  Passwordhelper,
  mode,
  error,
  email,
  password,
}) => {
  const onClick = () => {
    toast.error('아이디와 비밀번호가 비었습니다!', {
      position: 'bottom-right',
      autoClose: 5000,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,

      type: 'error',
    });
  };

  React.useEffect(() => {
    const isUserInputError = error?.graphQLErrors?.map(e => e.message);
    if (error) {
      toast.error(
        isUserInputError.length > 0 ? isUserInputError[0] : '서버에 문제가 생겼습니다',
        {
          position: 'bottom-right',
          autoClose: 5000,
          hideProgressBar: false,
          closeOnClick: true,
          pauseOnHover: true,
          draggable: true,
          progress: undefined,

          type: 'error',
        },
      );
    }
  }, [error]);

  //  labelPlaceholder="Email"
  return (
    <>
      <div className="px-[6.46875rem] py-[1.5rem] mmd:px-[2rem]">
        <InputPlaceHolder className="flex items-center">
          <Input
            {...EB}
            clearable
            status={helper?.color}
            color={helper?.color}
            helperColor={helper?.color}
            helperText={helper?.text}
            width="100%"
            className="w-full"
            type="email"
            placeholder="Email"
          />
        </InputPlaceHolder>
        <InputPlaceHolder className="flex items-center mt-12">
          <Input.Password
            {...PB}
            clearable
            status={Passwordhelper?.color}
            color={Passwordhelper?.color}
            helperColor={Passwordhelper?.color}
            helperText={Passwordhelper?.text}
            className="w-full"
            type="password"
            placeholder="Password"
            width="100%"
          />
        </InputPlaceHolder>

        <motion.button
          onClick={(e: any) => {
            helper.color == 'success' && Passwordhelper.color == 'success'
              ? handleSubmit(e)
              : null;
            !email && !password ? onClick() : '';
          }}
          whileHover={{ scale: 1.05 }}
          whileTap={{ scale: 0.8 }}
          className="flex bg-[#fcd435] text-[#212529] mt-12 h-12 justify-center items-center tracking-widest w-full rounded-xl">
          {auth}
        </motion.button>
      </div>
      <div className="px-[6rem]">
        {/* {mode == 'register' ? '회원가입' : '로그인'}으로 이동 */}
      </div>

      <div className="flex my-4 items-center justify-between px-1">
        <div className="w-[136px] h-[1px] bg-[#EAECEF]"></div>
        <div className="text-[#707a8a] font-Roboto">Or</div>
        <div className="w-[136px] h-[1px] bg-[#EAECEF]"></div>
      </div>

      <div className="flex justify-between w-[50%] mx-auto pt-3">
        <Google />
        <FaceBook />
      </div>
    </>
  );
};

export default React.memo(AuthForm);

const InputPlaceHolder = styled.div`
  .nextui-c-hzQjrs {
  }
`;
