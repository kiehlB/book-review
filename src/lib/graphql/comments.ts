import gql from 'graphql-tag';

export const CreateComment = gql`
  mutation CreateSub($post_id: String!, $text: String!, $comment_id: String) {
    createSub(post_id: $post_id, text: $text, comment_id: $comment_id) {
      id
      level
    }
  }
`;

export const RemoveSub = gql`
  mutation RemoveSub($id: String!) {
    removeSub(id: $id)
  }
`;

export const EditSub = gql`
  mutation EditSub($id: String!, $text: String!) {
    editSub(id: $id, text: $text) {
      id
      text
      level
      updated_at
      created_at
      updated_at
    }
  }
`;

export const GET_COMMENTS_COUNT = gql`
  query GetCommentsCount($id: String!) {
    post(id: $id) {
      id
      subs_count
    }
  }
`;

export const GET_SubComment = gql`
  query GetSub($comment_id: String!) {
    getSub(comment_id: $comment_id) {
      id
      text
      likes
      has_replies
      deleted
      upvotes
      likedSub
      users {
        id
        username
      }
      level
      reply
      created_at
      updated_at
      replies_count
      replies {
        id
        text
        level
        updated_at
        created_at
        has_replies
        replies_count
        upvotes
        user {
          id
          username
          profile {
            id
            bio
            profile_name
            thumbnail
          }
        }
      }
    }
  }
`;

export const CommentUpvotes = gql`
  mutation commentUpvotes($id: String!) {
    commentUpvotes(id: $id)
  }
`;

export const DeleteCommentUpvote = gql`
  mutation DeleteCommentUpvote($id: String!) {
    deleteCommentUpvote(id: $id)
  }
`;
