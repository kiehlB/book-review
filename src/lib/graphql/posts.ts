import gql from 'graphql-tag';

export const GET_Post = gql`
  query GetPost($id: String!) {
    post(id: $id) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      subs_count
      released_at
      created_at
      updated_at
      liked
      tags {
        tag {
          name
        }
      }
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
        likes
        has_replies
        deleted
        level
        reply
        created_at
        updated_at
        upvotes
        replies {
          id
          text
          updated_at
        }
        replies_count
        user {
          id
          username
          profile {
            id
            bio
            profile_name
          }
        }
      }
    }
  }
`;

export const GET_Posts = gql`
  query GetPosts($cursor: String, $username: String, $istemp: Boolean) {
    posts(cursor: $cursor, username: $username, istemp: $istemp) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      tags {
        tag {
          name
        }
      }
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const GET_recentPosts = gql`
  query RecentPosts($cursor: String, $limit: Int) {
    recentPosts(cursor: $cursor, limit: $limit) {
      id
      title

      thumbnail
      likes
      views
      is_temp
      postbody
      is_private
      released_at
      created_at
      updated_at
      liked
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const GET_trendingPosts = gql`
  query TrendingPosts($offset: Int, $limit: Int, $timeframe: String) {
    trendingPosts(offset: $offset, limit: $limit, timeframe: $timeframe) {
      id
      title

      thumbnail
      likes
      postbody
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const Create_Post = gql`
  mutation CreatePost(
    $body: String!
    $title: String!
    $thumbnail: String
    $postbody: String
    $tags: [String]
    $is_temp: Boolean
    $is_private: Boolean
    $bookTitle: String
    $bookContent: String
    $bookUrl: String
    $bookIsbn: String
    $bookAuthors: [String]
  ) {
    createPost(
      body: $body
      title: $title
      tags: $tags
      thumbnail: $thumbnail
      postbody: $postbody
      is_temp: $is_temp
      is_private: $is_private
      bookTitle: $bookTitle
      bookContent: $bookContent
      bookUrl: $bookUrl
      bookIsbn: $bookIsbn
      bookAuthors: $bookAuthors
    ) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      postbody
      is_private
      released_at
      created_at
      updated_at
      liked
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
    }
  }
`;

export const RELOAD_COMMENTS = gql`
  query ReloadComments($id: String!) {
    post(id: $id) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      subs_count
      released_at
      created_at
      updated_at
      liked
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
        updated_at
      }
    }
  }
`;

export const GET_Search_Posts = gql`
  query SearchPosts($searchInput: String) {
    searchPosts(searchInput: $searchInput) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      bookInfo {
        bookTitle
        bookContent
        bookUrl
        bookIsbn
        bookAuthors
      }
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const Edit_Post = gql`
  mutation EditPost(
    $id: String!
    $title: String
    $body: String
    $thumbnail: String
    $tags: [String]
    $is_temp: Boolean
    $is_private: Boolean
  ) {
    editPost(
      id: $id
      title: $title
      body: $body
      tags: $tags
      is_temp: $is_temp
      thumbnail: $thumbnail
      is_private: $is_private
    ) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const Remove_Post = gql`
  mutation RemovePost($id: String!) {
    removePost(id: $id)
  }
`;

export const Like_Post = gql`
  mutation LikePost($id: String!) {
    likePost(id: $id) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const UnLike_Post = gql`
  mutation UnlikePost($id: String!) {
    unlikePost(id: $id) {
      id
      title
      body
      thumbnail
      likes
      views
      is_temp
      is_private
      released_at
      created_at
      updated_at
      liked
      user {
        id
        username
      }
      subs {
        id
        text
      }
      subs_count
    }
  }
`;

export const UPLOAD_IMAGE_TO_CLOUDINARY = gql`
  mutation UploadImageToCloudinary($body: String!) {
    uploadImage(body: $body) {
      public_id
      url
    }
  }
`;
