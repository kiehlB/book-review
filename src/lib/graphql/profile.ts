import gql from 'graphql-tag';

export const createProfileMutation = gql`
  mutation CreateProfile($bio: String, $profile_name: String, $thumbnail: String) {
    createProfile(bio: $bio, profile_name: $profile_name, thumbnail: $thumbnail) {
      id
      bio
      profile_name
      thumbnail
      created_at
      updated_at
    }
  }
`;

export const getProfileQuery = gql`
  query GetProfile($user_id: String) {
    getProfile(user_id: $user_id) {
      id
      bio
      profile_name
      thumbnail
      created_at
      updated_at
    }
  }
`;
