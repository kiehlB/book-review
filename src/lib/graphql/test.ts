import gql from 'graphql-tag';

export const GetC = gql`
  query Me {
    me {
      c
    }
  }
`;

export const TC = gql`
  mutation Login($a: Int!, $b: Int!) {
    login(a: $a, b: $b) {
      c
    }
  }
`;
