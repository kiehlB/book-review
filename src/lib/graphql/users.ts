import gql from 'graphql-tag';

export const loginMutation = gql`
  mutation Login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      id
      accessToken
      username
    }
  }
`;

export const registerMutation = gql`
  mutation Register($email: String!, $password: String!) {
    register(email: $email, password: $password) {
      id
    }
  }
`;

export const getUsersQuery = gql`
  query Users {
    users {
      id
      username
      follower {
        id
        follower_id
      }
    }
  }
`;

export const meQuery = gql`
  query Me {
    me {
      id
      username
    }
  }
`;

export const logoutMutation = gql`
  mutation Logout {
    logout
  }
`;

export const followMutation = gql`
  mutation FollowUser($username: String!) {
    followUser(username: $username) {
      id
      user_id
      follower_id
    }
  }
`;

export const unFollowMutation = gql`
  mutation UnFollowUser($username: String!) {
    unFollowUser(username: $username) {
      id
    }
  }
`;
