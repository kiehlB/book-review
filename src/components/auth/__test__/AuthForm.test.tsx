import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import AuthForm, { AuthFormProps } from '../AuthForm';
import userEvent from '@testing-library/user-event';
import renderWithProviders from '../../../lib/renderWithProviders';
import { HelmetProvider } from 'react-helmet-async';
import { registerMutation } from '../../../lib/graphql/users';

describe('AuthForm', () => {
  const setup = (props: Partial<AuthFormProps> = {}) => {
    const initialProps = {
      username: '',
      password: '',
      handleChange: () => {},
      handleSubmit: () => {},
      mode: 'register',
      auth: 'Register',
      SetMode: () => {},
    };
    return render(<AuthForm {...initialProps} {...props} />);
  };

  it('renders correctly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe('handles modes', () => {
    it('REGISTER', () => {
      const { getByTestId } = setup();
      const title = getByTestId('title');
      expect(title).toHaveTextContent('Register');
    });
    it('LOGIN', () => {
      const { getByTestId } = setup({ auth: 'Login' });
      const title = getByTestId('title');
      expect(title).toHaveTextContent('Login');
    });
  });
});

describe('AuthForm', () => {
  const Passwordhelper = {
    color: 'success',
    state: 'idle',
    text: '',
  };

  const Usernamehelper = {
    color: 'success',
    state: 'idle',
    text: '',
  };

  it('renders correctly', () => {
    const { getByLabelText } = render(
      <AuthForm Passwordhelper={Passwordhelper} Usernamehelper={Usernamehelper} />,
    );

    const usernameInput = getByLabelText('username');
    const passwordInput = getByLabelText('password');

    fireEvent.change(usernameInput, { target: { value: 'johndoe' } });
    fireEvent.change(passwordInput, { target: { value: '1234' } });

    expect(usernameInput).toHaveValue('johndoe');
    expect(passwordInput).toHaveValue('1234');
  });

  it('handles form submission', () => {
    const handleSubmit = jest.fn();
    const { getByTestId } = render(
      <AuthForm
        Passwordhelper={Passwordhelper}
        Usernamehelper={Usernamehelper}
        handleSubmit={handleSubmit}
      />,
    );

    const submitButton = getByTestId('title');

    fireEvent.click(submitButton);

    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});

describe('Register', () => {
  const setup = (props = {}, overrideMocks?: any) => {
    const initialProps = {
      username: 'woong',
      password: 'sample',
      mode: 'register',
      auth: 'register',
      handleChange: () => {},
      handleSubmit: () => {},

      Passwordhelper: {
        color: 'success',
        state: 'idle',
        text: '',
      },
      Usernamehelper: {
        color: 'success',
        state: 'idle',
        text: '',
      },
    };

    const utils = renderWithProviders(
      <HelmetProvider>
        <AuthForm {...initialProps} {...props} />
      </HelmetProvider>,
      {
        mocks: overrideMocks,
      },
    );
    return {
      ...utils,
    };
  };
  it('renders post correctly', async () => {
    const mocks = [
      {
        request: {
          query: registerMutation,
          variables: {
            username: 'woong',
            password: 'sample',
          },
        },
        result: {
          data: {
            register: { username: 'woong' },
          },
        },
      },
    ];

    const { getByText, getByAltText } = setup({}, mocks);

    const button = await screen.findByText('register');
    userEvent.click(button);
    const registerResponse = mocks[0].result.data.register;
    expect(registerResponse.username).toEqual('woong');
  });
});
