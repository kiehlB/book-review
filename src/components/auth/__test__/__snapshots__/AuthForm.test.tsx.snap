// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm renders correctly 1`] = `
<div>
  <div
    class="px-[6.46875rem] py-[1.5rem] mmd:px-[2rem] ssm:px-[1rem]"
  >
    <div
      class="flex items-center"
    >
      <div
        class="form"
      >
        <input
          class="form__input"
          id="username"
          name="username"
          placeholder=" "
          type="username"
          value=""
        />
        <label
          class="form__label"
          for="username"
        >
          username
        </label>
      </div>
    </div>
    <div
      class="text-xs py-[0.5rem] px-1"
    />
    <div
      class="flex items-center mt-6"
    >
      <div
        class="form"
      >
        <input
          class="form__input border-[#d3d7e2]"
          id="password"
          name="password"
          placeholder=" "
          type="password"
          value=""
        />
        <label
          class="form__label"
          for="password"
        >
          password
        </label>
      </div>
    </div>
    <div
      class="text-xs py-[0.5rem] px-1"
    />
    <div>
      <button
        class="flex bg-[#fcd435] text-[#212529] mt-8 h-12 justify-center items-center tracking-widest w-full rounded-xl"
        data-testid="title"
      >
        Register
      </button>
    </div>
  </div>
  <div
    class="link px-[6.46875rem] text-base flex justify-end mmd:px-[2rem] ssm:px-[1rem]"
  >
    <div
      class="text-[#C99400] font-semibold cursor-pointer"
    >
      로그인
    </div>
    <div
      class="dark:text-[#e4e5e7]"
    >
       으로 이동
    </div>
  </div>
  <div
    class="flex mt-4  my-8 items-center justify-between px-1"
  >
    <div
      class="w-[136px] h-[1px] bg-[#EAECEF]"
    />
    <div
      class="text-[#707a8a] dark:text-[#e4e5e7]"
    >
      Or
    </div>
    <div
      class="w-[136px] h-[1px] bg-[#EAECEF]"
    />
  </div>
  <div
    class="flex justify-between w-[60%] mx-auto pt-3"
  >
    <a
      href="undefined"
    >
      <div
        class="rounded-full border-[#DEE2E6] border w-[50px] h-[50px] flex justify-center items-center transition-all hover:hover:translate-y-[-5px] transform duration-500 ease-in-out cursor-pointer hover:shadow-md"
      >
        <svg
          height="24px"
          viewBox="0 0 48 48"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#FFC107"
          />
          <path
            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
            fill="#FF3D00"
          />
          <path
            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
            fill="#4CAF50"
          />
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#1976D2"
          />
        </svg>
      </div>
    </a>
    <div>
      <div
        class="hidden"
        id="naverIdLogin"
      />
      <div
        class="rounded-full w-[50px] h-[50px] flex justify-center items-center transition-all hover:hover:translate-y-[-5px] transform duration-500 ease-in-out cursor-pointer hover:shadow-md"
      >
        <img
          src="/naver.png "
        />
      </div>
    </div>
    <a
      href="undefined"
    >
      <div
        class="rounded-full bg-[#fee500] w-[50px] h-[50px] flex justify-center items-center transition-all hover:hover:translate-y-[-5px] transform duration-500 ease-in-out cursor-pointer hover:shadow-md"
      >
        <img
          class="w-[24px] h-[24px]"
          src="/kakao.png"
        />
      </div>
    </a>
  </div>
</div>
`;
